### PR TITLE
Fixes for Web Request and HTTP handler + Web Request Test Case

### PR DESCRIPTION
**Pull request by GhadirCliqz**
@sam-cliqz @mahmoud-cliqz
1- Fixes for `HTTPhandler`: implemented a method that creates a directory in it doesn't exist while writing to a file.
2- Test case for blocking a web request after loading the Adblocker module.
3- Changes in the syntax of `webrequest.js` to solve the compatibility issues with `iOS 8, 9`. @sam-cliqz I hope it doesn't affect `Android`.
4- Minor changes in `WebRequest API` to solve problems with intercepting request.

### CODE REVIEW

File: jsengine-android/jsengine/src/main/assets/webrequest.js [hunk 1]
```diff
@@ -24,29 +24,33 @@ var webRequest = {
...
-          for (let listener of this.listeners) {
-              const {fn, filter, extraInfo} = listener;
-              const blockingResponse = fn(requestInfo);
-              if (blockingResponse && Object.keys(blockingResponse).length > 0) {
-                return blockingResponse;
-              }
-          }
-          return {};
+          var returnBlockingResponse = {};
+          this.listeners.forEach(function(listener) {
```
Comment (ghost) [inline jsengine-android/jsengine/src/main/assets/webrequest.js]: I think using a forEach here makes the logic a little messy. I imagine the reason for the change is a `for .. of` loop, but we can use a different kind of for loop instead, which will then look closer to the original implementation:

```
for (var i=0; i < this.listeners.length; i++) {
  const listener = this.listeners[i];
  ...
}
```

File: jsengine-ios/jsengine/SystemLoader.swift [hunk 2]
```diff
@@ -43,7 +43,9 @@ class SystemLoader {
         // some custom modules for the App: system and promise
         context.evaluateScript("System.set('system', { default: System });");
         context.evaluateScript("System.set('promise', { default: Promise });");
-
+        
+        // load webrequest
+        self.loadJavascriptSource("webrequest")
```
Comment (ghost) [inline jsengine-ios/jsengine/SystemLoader.swift]: shouldn't this be loaded in `WebRequest.swift`?

File: jsengine-android/jsengine/src/main/assets/webrequest.js [hunk 3]
```diff
@@ -29,18 +29,17 @@ var webRequest = {
           requestInfo.getRequestHeader = function(header) {
             return requestInfo.requestHeaders[header];
           };
-          var returnBlockingResponse = {};
-          this.listeners.forEach(function(listener) {
+          for (var i=0; i < this.listeners.length; i++) {
```
Comment (GhadirCliqz) [inline jsengine-android/jsengine/src/main/assets/webrequest.js]: @sam-cliqz here are the changes you requested.

File: jsengine-ios/jsengine/WebRequest.swift [hunk 4]
```diff
@@ -28,7 +29,14 @@ public class WebRequest {
         }
         context.setObject(unsafeBitCast(nativeIsWindowActive, AnyObject.self), forKeyedSubscript: "_nativeIsWindowActive")
         
-        self.webRequest = context.evaluateScript("webRequest")
+        if let content = Utils.readSourceFile(self.bundle, assetsRoot: "assets", assetPath: "webrequest", buildPath: "", fileExtension: "js") {
```
Comment (GhadirCliqz) [inline jsengine-ios/jsengine/WebRequest.swift]: @sam-cliqz I updated `WebRequest`, so that `webrequest.js` is loaded here.

**Pull request merged**